stdlib/LinearAlgebra/src/qr.jl

```diff
@@ -751,7 +751,19 @@ function *(adjA::Adjoint{<:Any,<:StridedVecOrMat}, adjQ::Adjoint{<:Any,<:Abstrac
 end
 
 ### mul!
-mul!(C::StridedVecOrMat{T}, Q::AbstractQ{T}, B::StridedVecOrMat{T}) where {T} = lmul!(Q, copyto!(C, B))
+function mul!(C::StridedVecOrMat{T}, Q::AbstractQ{T}, B::StridedVecOrMat{T}) where {T}
+    require_one_based_indexing(C, B)
+    mB = size(B, 1)
+    mC = size(C, 1)
+    if mB < mC
+        fill!(C, zero(T))
```

Comment:
This may be slightly wasteful, but I couldn't find how to elegantly fill only the complement of `C[eachindex(IndexCartesian(), B)]` with zeros.

Reply:
I guess explicit loops could work, since the dimensionality is fixed. Alternatively:
```julia
C[CartesianIndices(B)] .= B
C[CartesianIndices((mB+1:mC, 1:size(C,2))] .= zero(T)
```